src/main/java/core/basesyntax/service/impl/CsvReaderServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.AbstractTransaction;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.service.CsvReaderService;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+
+public class CsvReaderServiceImpl implements CsvReaderService<Fruit> {
+    private static final String DELIMITER = ",";
+    private static final int INDEX_OF_OPERATION_TYPE = 0;
+    private static final int INDEX_OF_FRUIT = 1;
+    private static final int INDEX_OF_COUNT = 2;
+
+    @Override
+    public List<AbstractTransaction<Fruit>> parse(String filePath) {
+        return readLines(filePath).stream().skip(1).map(this::parseLine).toList();
+    }
+
+    private List<String> readLines(String filePath) {
+        try {
+            FileReader fileReader = new FileReader(filePath);
+            try (BufferedReader bufferedReader = new BufferedReader(fileReader)) {
```

Comment:
The `FileReader` should be included in the try-with-resources statement to ensure it is closed properly. This prevents potential resource leaks. Consider using `try (FileReader fileReader = new FileReader(filePath); BufferedReader bufferedReader = new BufferedReader(fileReader)) { ... }`.